pymongo/command_cursor.py

```diff
@@ -113,7 +114,7 @@ def __die(self, synchronous: bool = False) -> None:
         self.__sock_mgr = None
 
     def __end_session(self, synchronous: bool) -> None:
-        if self.__session and not self.__explicit_session:
+        if self.__session and not self.__explicit_session and get_thread_client_session() is None:
```

Comment:
Why do we need to prevent `self.__session` from ending? It's a separate session from `thread_client_session`. If it was supposed to end, why not let it?

Reply:
its not separate if we injected the thread session, it get set higher up the stack